transfer_nlp/plugins/config.py

```diff
@@ -204,12 +204,18 @@ def recursive_replace(my_item: Union[Dict, List, str]):
 class ExperimentConfig:
 
     @staticmethod
-    def load_experiment_json(experiment: Union[str, Path, Dict]) -> Dict:
+    def load_experiment_dict(experiment: Union[str, Path, Dict]) -> Dict:
+        config = {}
         if isinstance(experiment, dict):
             config = dict(experiment)
         else:
             with open(experiment) as f:
-                config = json.load(f)
+                if str(experiment).endswith(".json") or str(experiment).endswith(".yaml") or str(experiment).endswith(".yml"):
```

Comment:
To avoid repeting yourself, you could convert experiment to a Path, then use suffix attribute
```python
experiment_path = Path(str(experiment)).expanduser()
with experiment_path.open() as f:
  if experiment_path.suffix in {'.jon', '.yaml', '.yml'}:
    ...
  elif ... :
    ...
```

Reply:
alright!